### PR TITLE
Add workflow to update the website for the book automatically

### PR DESCRIPTION
**Pull request by grief8**
This pull request allows updating the book once the `docs/` is updated. However, another setting should be applied to `asterinas/asterinas`:

- Enter `Settings -> Pages -> Build and deployment -> Branch`. 
- Select `gh-pages` and `Save`.
- Then you can access the book through `https://asterinas.github.io/asterinas/`.

### CODE REVIEW

**Comment by tatetian** [comment]
The Book repo is already hosted in [https://github.com/asterinas/book](https://github.com/asterinas/book) and accessible on https://asterinas.github.io/book. I would like to keep this arrangement of have a separate repo, instead of a branch, for the generated content by MdBook. Is it possible to update the asterinas/book repo, instead of the gh-pages branch of the asterinas/asterinas repo?

**Comment by grief8** [comment]
Of course we can still use `asterinas/book` repo to host `https://asterinas.github.io/book/`. The latest PR #683 implements such a workflow @tatetian . However, you also need to configure `asterinas/book` and `asterinas/asterinas`:

1. **Generate SSH key pair**: Open a terminal on your local computer and run the following command to generate a new SSH key pair. `ssh-keygen -t rsa -b 4096 -C "github-actions[bot]@users.noreply.github.com"`
2. **Add the public key to `asterinas/book` repository**: Open the public key file and copy its contents. Then, on GitHub, navigate to the `Settings -> Deploy keys` section, click `Add deploy key`, paste your public key, and give it a clear title like `DEPLOY_KEY_PUB`. Make sure "**Allow write access**" is checked so that deployment scripts can push to this repository.
3. **Add the private key to the Secrets of `asterinas/asterinas` repository**: Next, you need to add the private key to the Secrets of the source repository. First open the private key file containing the private key and copy its contents. On GitHub, navigate to the `Settings -> Secrets` section of `asterinas/asterinas` and click `Secrets and variables -> Actions -> New repository secret`. Give this secret a name (e.g. **DEPLOY_KEY**) and paste the contents of your private key. Click `Add secret` to save.

After the steps above, change any contents below `asterinas/docs/src/` and push the commits to `main`, you will see the changes later in `asterinas/book` or `https://asterinas.github.io/book/`.

File: .github/workflows/update-website.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          chmod 600 ./deploy_key
+          ssh-keyscan -t rsa github.com >> ./known_hosts
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          GIT_SSH_COMMAND="ssh -i ./deploy_key -o UserKnownHostsFile=./known_hosts" git clone git@github.com:asterinas/book.git
+          cd book
+          git checkout main
+          cp -r ../asterinas/docs/book/* .
+          git add .
+          git commit -m "Update website"
```
Comment (tatetian) [inline .github/workflows/update-website.yml:40]: Adding a new commit is ok but not ideal. First, keeping the entire history of the generated content of MdBook is of no use.  The source of the Book is version controlled by the `asterinas/asterinas` repo. That's what matters. The latest content of `asterinas/book` matters, but its past versions are not. Second, the command of `cp -r ../asterinas/docs/book/* .` can only add new files or updates existing files, but not delete old files.

File: .github/workflows/update-website.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          path: 'asterinas'
+      
+      - name: Build the website
+        run: |
+          cd asterinas/docs
+          mdbook build
+
+      - name: Deploy website
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
```
Comment (tatetian) [inline .github/workflows/update-website.yml:28]: `BOOK_PUBLISH_KEY` looks a better name than `DEPLOY_KEY` as the former name is more specific than the latter one.

**Review by tatetian — COMMENTED** [review]
Thanks for the detailed instruction!

**Comment by tatetian** [comment]
The secret key has been set up. Use the `BOOK_PUBLISH_KEY` secret variable to access it in CI.

File: .github/workflows/update-website.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Update Website
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/src/**'
```
[the remark this answers is not in the document]
Comment (grief8) [reply .github/workflows/update-website.yml:8]: You're right!

File: .github/workflows/update-website.yml (hunk 1 above)
Comment (grief8) [reply .github/workflows/update-website.yml:40]: I understand the point of this, I will try to modify the workflow to accomplish this.

**Comment by grief8** [comment]
> The secret key has been set up. Use the `BOOK_PUBLISH_KEY` secret variable to access it in CI.

Great! I will update the workflow soon.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**